### ISSUE TITLE
Option to choose home view: inbox, recent conversations, or all messages

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#517]
This is a followup issue to:
* #516

Specifically, it'd be good to let the user choose whether the screen the app launches on is "Inbox", "Recent conversations" (after #315), or "All messages".

We have a similar setting in web: https://chat.zulip.org/help/configure-home-view . It's expressed as `web_home_view` [in the register-queue response](https://zulip.com/api/register-queue).

It's possible the right answer here is that we should just follow the same setting, despite its name (and even though we [just renamed it](https://chat.zulip.org/#narrow/stream/378-api-design/topic/renaming.20.60default_view.60.20setting/near/1665385) to that a few months ago). As I think about it now, the choice between these is naturally driven mainly by the level of traffic in the given realm and the user's habits of keeping unreads clear or not (the factors we also cite in [our advice in the Help Center](https://chat.zulip.org/help/configure-home-view)). That means:
* It'll naturally vary for a given user between different realms they participate in, so that it doesn't fit well as a per-client setting along the lines of "open links in app or in external browser".
* It seems like it'll naturally align across devices for a given user in a given realm. After all, they see the same set of unreads on any device.

**Comment by gnprice** [comment]
This is closely related to:
* #311

**Comment by c-o-d-e-y-o-d-a** [comment]
Hello! I'd like to work on this issue. My proposed solution is to first ask the user for their preferred default screen when the app is opened for the first time. Additionally, I will implement a dropdown option in the settings where users can easily change their default page selection at any time. Let me know if this approach sounds good, and I'd be happy to proceed with the implementation!

**Comment by chrisbobbe** [comment]
My view is that this issue isn't yet ripe to work on. It's marked post-launch, so we'll have little to no time available to review work on it before the app launches.

**Comment by c-o-d-e-y-o-d-a** [comment]
Understandable, I will try to contribute somewhere else, thanks.

**Comment by gnprice** [comment]
Possibly a fourth option to offer for the same setting: the channels page, after we do
* #412.

(Or that'd make a third option, if we do this before implementing #315 "Recent conversations".)

I think that'd pretty much do what [this user request today](https://chat.zulip.org/#narrow/channel/48-mobile/topic/inbox.20and.20alternatives/near/2003132) asks for (more details at the link):
> IMO the mobile app's home screen should basically be the left panel on the desktop app, with Channels as the primary form of the content, and options for inbox/combined feed/mentions/etc as navigation options from there.

**Comment by chrisbobbe** [comment]
Related:

- #1164

**Comment by chrisbobbe** [comment]
We had a request for this in early launch feedback: [#mobile > Flutter App: Switch Inbox to different view? @ 💬](https://chat.zulip.org/#narrow/channel/48-mobile/topic/Flutter.20App.3A.20Switch.20Inbox.20to.20different.20view.3F/near/2203090)
